dotCMS/src/main/java/com/dotcms/cms/login/LoginServiceAPIImpl.java

```diff
@@ -0,0 +1,537 @@
+                    throws NoSuchUserException {
+
+        return _doLogin(userName, password, rememberMe, request, response, false);
+    }
+
+
+
+    private boolean _doLogin(final String userIdOrEmail, final String password,
+                    final boolean rememberMe, final HttpServletRequest request,
+                    final HttpServletResponse response, final boolean skipPasswordCheck)
+                    throws NoSuchUserException {
+
+        LoginEvent event = new LoginEvent(this, LoginEvents.PRE_LOGIN, userIdOrEmail, password,
+                        request, response);
+        for (LoginEventListener listener : _listeners) {
+            listener.loginEventReceived(event);
+        }
+
+        boolean auth = _doLoginInternal(userIdOrEmail, password, rememberMe, request, response,
+                        false);
+
+        event = (auth) ? new LoginEvent(this, LoginEvents.POST_LOGIN_SUCCESS, userIdOrEmail,
+                        password, request, response)
+                        : new LoginEvent(this, LoginEvents.POST_LOGIN_FAILURE, userIdOrEmail,
+                                        password, request, response);
+        for (LoginEventListener listener : _listeners) {
+            listener.loginEventReceived(event);
+        }
+
+        return auth;
+
+
+    }
+
+
+
+    private boolean _doLoginInternal(final String userIdOrEmail, final String password,
+                    final boolean rememberMe, final HttpServletRequest request,
+                    final HttpServletResponse response, final boolean skipPasswordCheck)
+                    throws NoSuchUserException {
+        Company company = PublicCompanyFactory.getDefaultCompany();
+        User user = null;
+
+        try {
+
+
+
+            // if skipPasswordCheck is in true, means we do not have to check the user on
+            // LDAP just our DB
+            if (!skipPasswordCheck) {
+                if (runPreAuth(userIdOrEmail, password, company) == Authenticator.SUCCESS) {
+                    syncPassword(_loadUserByKey(userIdOrEmail));
+                }
+            }
+
+            user = _loadUserByKey(userIdOrEmail);
+
+            if (!skipPasswordCheck) {
+                passwordMatch(password, user);
+            }
+
+
+
+            doRememberMe(request, response, user, rememberMe);
+
+
+            HttpSession ses = request.getSession();
+
+
+            ses.setAttribute(com.dotmarketing.util.WebKeys.CMS_USER, user);
+            ses.setAttribute(WebKeys.USER_ID, user.getUserId());
+            SecurityLogger.logInfo(this.getClass(), "User " + userIdOrEmail
+                            + " has sucessfully login from IP: " + request.getRemoteAddr());
+
+
+            return true;
+
+        } catch (NoSuchUserException e) {
+            Logger.error(this.getClass(), "User " + userIdOrEmail + " does not exist.", e);
+            SecurityLogger.logInfo(this.getClass(), "An invalid attempt to login as "
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "An invalid attempt to login as " 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)